components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/utils/ServiceCatalogUtils.java

```diff
@@ -655,4 +754,64 @@ private static String getFileChecksum(File file) throws NoSuchAlgorithmException
         }
         return sb.toString();
     }
+
+    /**
+     * Reads WSDL of the proxy service and make a WSDL file in the given location.
+     *
+     * @param metadataYamlFile service metadata yaml file.
+     * @param storeLocation read wsdl file store location.
+     * @return WSDL file creation result.
+     */
+    private static boolean readProxyServiceWSDL(File metadataYamlFile, File storeLocation) {
+        try {
+            String proxyServiceUrl = getProxyServiceUrlFromMetadata(metadataYamlFile);
+            if (proxyServiceUrl == null) {
+                return false;
+            }
+            if (proxyServiceUrl.endsWith(PATH_SEPARATOR)) {
+                proxyServiceUrl = proxyServiceUrl.substring(proxyServiceUrl.length() - 1);
+            }
+            proxyServiceUrl = proxyServiceUrl + WSDL_URL_PATH;
+            URL website = new URL(proxyServiceUrl);
+            URLConnection connection = website.openConnection();
+            BufferedReader in = new BufferedReader(new InputStreamReader(connection.getInputStream()));
+            StringBuilder responseWSDL = new StringBuilder();
+            String inputLine;
+            while ((inputLine = in.readLine()) != null) {
+                responseWSDL.append(inputLine).append(System.getProperty("line.separator"));
+            }
+            in.close();
+
+            if (responseWSDL.length() == 0) {
+                return false;
+            }
+            if (storeLocation.exists()) {
+                Files.write(Paths.get(storeLocation.getAbsolutePath(), WSDL_FILE_NAME),
+                        responseWSDL.toString().getBytes());
+                return true;
+            }
+        } catch (IOException e) {
+            log.error("Error while getting WSDL definition of the given proxy service", e);
+        }
+
+        return false;
+    }
+
+    /**
+     * Reads service URL from metadata file.
+     *
+     * @param yamlFile location of the yamlFile.
+     * @return URL of the metadata service.
+     */
+    private static String getProxyServiceUrlFromMetadata(File yamlFile) {
+        Yaml yaml = new Yaml();
+        String currentServiceUrl = null;
+        try (InputStream yamlStream = new FileInputStream(yamlFile)) {
+            Map<String, Object> obj = (Map<String, Object>) yaml.load(yamlStream);
+            currentServiceUrl = (String) obj.get(SERVICE_URL);
```

Comment:
We only support 1 service URL, is it? There's no way for us to expose 2 for http and https?

Reply:
yes, the Service catalog is designed to use one service URL from the metadata file.

Reply:
@tharindu1st Shouldn't this option be available?

Reply:
As per the offline chat with Tharindu this will not be required.